bootstrap/file/file_test.go

```diff
@@ -0,0 +1,58 @@
+package file
+
+import (
+	"errors"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/container"
+	"github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/interfaces/mocks"
+	"github.com/edgexfoundry/go-mod-bootstrap/v3/di"
+	"github.com/stretchr/testify/assert"
+)
+
+var dic *di.Container
+
+func TestLoadFile(t *testing.T) {
+	tests := []struct {
+		Name               string
+		Path               string
+		ContentLength      int
+		ExpectedErr        string
+		expectedSecretData map[string]string
+	}{
+		{"Valid - load from YAML file", path.Join("..", "config", "testdata", "configuration.yaml"), 4533, "", nil},
+		{"Valid - load from JSON file", path.Join(".", "testdata", "configuration.json"), 142, "", nil},
+		{"Valid - load from HTTPS", "https://raw.githubusercontent.com/edgexfoundry/go-mod-bootstrap/main/bootstrap/config/testdata/configuration.yaml", 4533, "", nil},
+		{"Invalid - File not found", "bogus", 0, "Could not read file", nil},
+		{"Invalid - load from invalid HTTPS", "https://raw.githubusercontent.com/edgexfoundry/go-mod-bootstrap/main/bootstrap/config/configuration.yaml", 1, "Invalid status code", nil},
+		{"valid - load from HTTPS with secret", "https://raw.githubusercontent.com/edgexfoundry/go-mod-bootstrap/main/bootstrap/config/testdata/configuration.yaml?edgexSecretName=mySecretName", 4533, "", map[string]string{"type": "httpheader", "headername": "Authorization", "headercontents": "Basic 1234567890"}},
```

Comment:
Can we move this test case up to where the other valid ones are?
```suggestion
		{"Valid - load from HTTPS with secret", "https://raw.githubusercontent.com/edgexfoundry/go-mod-bootstrap/main/bootstrap/config/testdata/configuration.yaml?edgexSecretName=mySecretName", 4533, "", map[string]string{"type": "httpheader", "headername": "Authorization", "headercontents": "Basic 1234567890"}},
```